src/main/java/uk/gov/hmcts/reform/cpo/controllers/CasePaymentOrdersController.java

```diff
@@ -51,25 +50,70 @@ public CasePaymentOrdersController(CasePaymentOrdersService casePaymentOrdersSer
+            code = 400,
+            message = "One or more of the following reasons:"
+                + "\n1) " + ValidationError.CPO_FILER_ERROR
+                + "\n2) " + ValidationError.CASE_ID_INVALID
+                + "\n3) " + ValidationError.ID_INVALID,
+            response = String.class,
+            examples = @Example({
+                @ExampleProperty(
+                    value = "{\n"
+                        + "   \"status\": \"400\",\n"
+                        + "   \"error\": \"Bad Request\",\n"
+                        + "   \"message\": \"" + ValidationError.ARGUMENT_NOT_VALID + "\",\n"
+                        + "   \"path\": \"" + CASE_PAYMENT_ORDERS_PATH + "\",\n"
+                        + "   \"details\": [ \""  + ValidationError.CASE_ID_INVALID + "\" ]\n"
+                        + "}",
+                    mediaType = APPLICATION_JSON_VALUE)
+            })
+        ),
+        @ApiResponse(
+            code = 401,
+            message = AuthError.AUTHENTICATION_TOKEN_INVALID
+        ),
+        @ApiResponse(
+            code = 403,
+            message = AuthError.UNAUTHORISED_S2S_SERVICE
+        )
+    })
+    public Page<CasePaymentOrder> getCasePaymentOrders(@ApiParam("list of case payment orders ids")
+                                                       @ValidCpoId
+                                                       @RequestParam(name = "ids", required = false)
+                                                       Optional<List<String>> ids,
+                                                       @ApiParam("list of ccd case reference numbers")
+                                                       @ValidCaseId
+                                                       @RequestParam(name = "cases-ids", required = false)
+                                                       Optional<List<String>> casesId,
+                                                       @RequestParam(name = "pageSize", required = false)
+                                                       Optional<Integer> pageSize,
+                                                       @RequestParam(name = "pageNumber", required = false)
+                                                       Optional<Integer> pageNumber
 
     ) {
+
         final List<String> listOfIds = ids.orElse(Collections.emptyList());
         final List<String> listOfCasesIds = casesId.orElse(Collections.emptyList());
         final CasePaymentOrderQueryFilter casePaymentOrderQueryFilter = CasePaymentOrderQueryFilter.builder()
-            .listOfIds(listOfIds)
-            .listOfCasesIds(listOfCasesIds)
-            .pageNumber(pageNumber.orElse(Integer.parseInt(applicationParams.getDefaultPageNumber())))
-            .pageSize(pageSize.orElse(Integer.parseInt(applicationParams.getDefaultPageSize())))
+            .cpoIds(listOfIds)
+            .caseIds(listOfCasesIds)
+            .pageNumber(pageNumber.orElse(CasePaymentOrderQueryFilter.PAGE_NUMBER))
```

Comment:
2 things:
1) Do we really need to do this? I think Sprint already uses 0 by default if no page number is specified isn't it?
2) you can have the `pageNumber` itself do: 
`pageNumber.orElse(CasePaymentOrderQueryFilter.PAGE_NUMBER)`
this way `CasePaymentOrderQueryFilter.PAGE_NUMBER` is used only internally by `CasePaymentOrderQueryFilter`

Reply:
TODO

Reply:
Mario I had a look  to AbstractPageRequest and it looks like that the page it has to be always set .